addons/portal/static/src/js/portal_name_and_signature.js

```diff
@@ -432,6 +434,25 @@ var NameAndSignature = Widget.extend({
             }
         }, 0);
     },
+    /**
+     * Waits for the signature to be not empty and triggers up the event
+     * `signature_changed`.
+     * This is necessary because some methods of jSignature are async but
+     * they don't return a promise and don't trigger any event.
+     *
+     * @private
+     * @returns {Deferred}
+     */
+    _waitForSignatureNotEmpty: function () {
+        var def = $.Deferred();
```

Comment:
The deferred is created at each loop turn, to be perfect I would put the deferred in parameters.